src/PrestaShopBundle/Twig/Component/Toolbar.php

```diff
@@ -28,19 +28,45 @@
 
 namespace PrestaShopBundle\Twig\Component;
 
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Twig\Layout\MenuBuilder;
 use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
 
 #[AsTwigComponent(template: '@PrestaShop/Admin/Component/Layout/toolbar.html.twig')]
 class Toolbar
 {
     public bool $lockedToAllShopContext;
-    public array $breadcrumbs2;
     public array $toolbarBtn;
     public array|string $title;
     public string $table;
     public bool|string $helpLink;
     public bool $enableSidebar;
     public int $currentTabLevel;
     public array $tabs;
+    public array $breadcrumbs;
     public bool $useRegularH1Structure = true;
+
+    public function __construct(
+        private readonly HookDispatcherInterface $hookDispatcher,
+        private readonly MenuBuilder $menuBuilder
+    ) {
+    }
+
+    public function mount(): void
+    {
+        $tab = $this->menuBuilder->getCurrentTab();
+        if (null === $tab) {
+            $this->breadcrumbs = [];
+        } else {
+            $tabs = [];
+            $tabs[] = $tab;
+            $ancestorsTab = $this->menuBuilder->getAncestorsTab($tab->getId());
+            if (!empty($ancestorsTab)) {
+                $tabs[] = $ancestorsTab;
+            }
+
+            $this->breadcrumbs = $this->menuBuilder->convertTabsToBreadcrumbLinks($tab, $ancestorsTab);
+            $this->hookDispatcher->dispatchWithParameters('actionAdminBreadcrumbModifier', ['tabs' => $tabs, 'breadcrumb' => &$this->breadcrumbs]);
+        }
+    }
```

Comment:
```suggestion
    public array $breadcrumbs = [];
    public bool $useRegularH1Structure = true;

    public function __construct(
        private readonly HookDispatcherInterface $hookDispatcher,
        private readonly MenuBuilder $menuBuilder
    ) {
    }

    public function mount(): void
    {
        $tab = $this->menuBuilder->getCurrentTab();
        if ($tab) {
            $tabs = [];
            $tabs[] = $tab;
            $ancestorsTab = $this->menuBuilder->getAncestorsTab($tab->getId());
            if (!empty($ancestorsTab)) {
                $tabs[] = $ancestorsTab;
            }

            $this->breadcrumbs = $this->menuBuilder->convertTabsToBreadcrumbLinks($tab, $ancestorsTab);
            $this->hookDispatcher->dispatchWithParameters('actionAdminBreadcrumbModifier', ['tabs' => $tabs, 'breadcrumb' => &$this->breadcrumbs]);
        }
    }
```

Reply:
Thanks for the feedback :) I made the same changes on another branch, based on this one and continuing the migration.